src/components/NavBar/SearchBar.tsx

```diff
@@ -201,61 +211,91 @@ export const SearchBarDropdown = ({ toggleOpen, tokens, collections, hasInput }:
     }
   }, [toggleOpen, hoveredIndex, totalSuggestions])
 
+  useEffect(() => {
```

Comment:
instead of doing this and storing the whole rendered state in a node - is there something wrong with cacheing `tokens` and `collections` in a local variable and updating those in a `useEffect` and you can render the state normally in the `render` method?

Reply:
That was my original attempt at a solution however I ran into two issues when trying that. 
1. You also need to be able to fade out on the default search results dropdown, ie search history, trending tokens, and trending collections, so tracking the old state for that data adds a layer of complexity.
2. When I attempted to useRef for tracking token and collection search state, when interacting with useQuery, the final re-render would never occur. So I would enter a search query, default results will fade, but then unfade without showing the new results. Then when next text was entered, the expected results from the previous query would render.

Reply:
ah ok figured there was reasoning to it